### PR TITLE
Several modifications

### PR DESCRIPTION
**Pull request by AndraB90**
Added linear search algorithms, 
restructured the packages,
added isSorted method to all sorting algorithms

### CODE REVIEW

File: src/search_algorithms/linear_search/GuessTheNumberGame.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package search_algorithms.linear_search;
+
+import java.util.Random;
+import java.util.Scanner;
+
+public class GuessTheNumberGame {
+    public static void main(String[] args) {
+        System.out.println("Input a number between 1 and 100: ");
+        int[] computerArr = new int[100];
+        for (int i = 1; i <= 100; i++) {
```
Comment (CorlyBoy) [inline src/search_algorithms/linear_search/GuessTheNumberGame.java:10]: please insert a new line before procedures (for, if, while, etc)

File: src/search_algorithms/linear_search/GuessTheNumberGame.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        Random r = new Random();
+        int secretNumber = r.nextInt(1, 101);
+        int playerGuess;
+        int computerGuess;
+        int count = 0;
+
+        while (true) {
+            playerGuess = scan.nextInt();
+            count++;
+            if (playerGuess < secretNumber) {
```
Comment (CorlyBoy) [inline src/search_algorithms/linear_search/GuessTheNumberGame.java:23]: same as above

File: src/search_algorithms/linear_search/GuessTheNumberGame.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                System.out.println("More");
+            } else if (playerGuess > secretNumber) {
+                System.out.println("less");
+            } else {
+                System.out.println("You win");
+                System.out.println("You used " + count + " trials");
+                break;
+            }
+        }
+
```
Comment (CorlyBoy) [inline src/search_algorithms/linear_search/GuessTheNumberGame.java:33]: Unnecessary extra line

File: src/search_algorithms/linear_search/GuessTheNumberGame.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        int computerGuessIndex = LinearSearch.findNumber(computerArr, secretNumber);
+        if (computerGuessIndex != -1) {
+            System.out.println("Computer's guess: " + computerArr[computerGuessIndex]);
+        } else {
+            System.out.println("Computer did not find the number.");
+        }
+
```
Comment (CorlyBoy) [inline src/search_algorithms/linear_search/GuessTheNumberGame.java:40]: same as above

File: src/sorting_algorithms/bubble_sort/Example1.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+import timer.StopWatch;
+
+import java.util.Random;
+
+public class Example1 {
+    public static void main(String[] args) {
+        int[] array = new int[100_000];
+        // 30.072 seconds * 10 * O(n) for 1 million elements
+        Random generator = new Random();
+        for (int i = 0; i < array.length; i++) {
```
Comment (CorlyBoy) [inline src/sorting_algorithms/bubble_sort/Example1.java:12]: same as above

File: src/sorting_algorithms/selection_sort/Example1.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import timer.StopWatch;
+
+import java.util.Random;
+
+public class Example1 {
+    public static void main(String[] args) {
+        int[] array = new int[100_000];
+        // 2.842 * 10 * O(n) seconds for 1 million elements
+        Random generator = new Random();
+        for (int i = 0; i < array.length; i++) {
```
Comment (CorlyBoy) [inline src/sorting_algorithms/selection_sort/Example1.java:12]: same as above

File: src/sorting_algorithms/selection_sort/Example1.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        StopWatch watch = new StopWatch();
+        System.out.println("Before sorting: ");
+        //printArray(array);
+        watch.start();
+        selectionSort(array);
+        System.out.println("After sorting: ");
+        //printArray(array);
+        watch.stop();
+        watch.printElapsedTime();
+
```
Comment (CorlyBoy) [inline src/sorting_algorithms/selection_sort/Example1.java:25]: same as above

File: src/search_algorithms/linear_search/GuessTheNumberGame.java (hunk 1 above)
Comment (AndraB90) [reply src/search_algorithms/linear_search/GuessTheNumberGame.java:10]: Resolved

File: src/search_algorithms/linear_search/GuessTheNumberGame.java (hunk 2 above)
Comment (AndraB90) [reply src/search_algorithms/linear_search/GuessTheNumberGame.java:23]: Done

File: src/search_algorithms/linear_search/GuessTheNumberGame.java (hunk 3 above)
Comment (AndraB90) [reply src/search_algorithms/linear_search/GuessTheNumberGame.java:33]: Resolved

File: src/search_algorithms/linear_search/GuessTheNumberGame.java (hunk 4 above)
Comment (AndraB90) [reply src/search_algorithms/linear_search/GuessTheNumberGame.java:40]: Done

File: src/sorting_algorithms/selection_sort/Example1.java (hunk 6 above)
Comment (AndraB90) [reply src/sorting_algorithms/selection_sort/Example1.java:12]: Done

File: src/sorting_algorithms/selection_sort/Example1.java (hunk 7 above)
Comment (AndraB90) [reply src/sorting_algorithms/selection_sort/Example1.java:25]: Same as above, resolved

**Review by CorlyBoy — APPROVED** [review]
Looks good!

**Pull request merged**